### ISSUE TITLE
Support Python 3.13

### ISSUE THREAD

**Issue by febarbosa182** [messense/py-promql-parser#13]
Output error when using python 3.13:

```
Building wheels for collected packages: promql-parser
  Building wheel for promql-parser (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building wheel for promql-parser (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [71 lines of output]
      Running `maturin pep517 build-wheel -i /Users/admin/projects/kepimetheus/venv/bin/python3 --compatibility off`
      📦 Including license file "/private/var/folders/_y/z8sthm2j7s9f5rbld7j1xbq00000gp/T/pip-install-_5ecfb_4/promql-parser_59f88903cc0b4cb88976d66a9e022747/LICENSE"
      🔗 Found pyo3 bindings
      🐍 Found CPython 3.13 at /Users/admin/projects/kepimetheus/venv/bin/python3
      💻 Using `MACOSX_DEPLOYMENT_TARGET=11.0` for aarch64-apple-darwin by default
         Compiling proc-macro2 v1.0.85
         Compiling autocfg v1.3.0
         Compiling unicode-ident v1.0.12
         Compiling libc v0.2.155
         Compiling serde v1.0.203
         Compiling semver v1.0.23
         Compiling rustversion v1.0.17
         Compiling powerfmt v0.2.0
         Compiling anyhow v1.0.86
         Compiling deranged v0.3.11
         Compiling num-traits v0.2.19
         Compiling num-conv v0.1.0
         Compiling itoa v1.0.11
         Compiling time-core v0.1.2
         Compiling target-lexicon v0.12.14
         Compiling quote v1.0.36
         Compiling rustc_version v0.4.0
         Compiling syn v2.0.66
         Compiling vob v3.0.3
         Compiling memchr v2.7.4
         Compiling num_threads v0.1.7
         Compiling time v0.3.36
         Compiling aho-corasick v1.1.3
         Compiling pyo3-build-config v0.21.2
         Compiling once_cell v1.19.0
         Compiling regex-syntax v0.8.4
         Compiling vergen v8.3.1
         Compiling hashbrown v0.14.5
         Compiling equivalent v1.0.1
         Compiling indexmap v2.2.6
         Compiling lazy_static v1.4.0
         Compiling cfg-if v1.0.0
         Compiling lrpar v0.13.7
         Compiling lrlex v0.13.7
         Compiling fnv v1.0.7
         Compiling filetime v0.2.23
         Compiling static_assertions v1.1.0
         Compiling cactus v1.0.7
         Compiling unicode-width v0.1.13
         Compiling regex-automata v0.4.7
         Compiling getopts v0.2.21
         Compiling pyo3-ffi v0.21.2
         Compiling lock_api v0.4.12
         Compiling parking_lot_core v0.9.10
         Compiling memoffset v0.9.1
      error: failed to run custom build command for `pyo3-ffi v0.21.2`

      Caused by:
        process didn't exit successfully: `/private/var/folders/_y/z8sthm2j7s9f5rbld7j1xbq00000gp/T/pip-install-_5ecfb_4/promql-parser_59f88903cc0b4cb88976d66a9e022747/target/release/build/pyo3-ffi-aad81bfc0eeea363/build-script-build` (exit status: 1)
        --- stdout
        cargo:rerun-if-env-changed=PYO3_CROSS
        cargo:rerun-if-env-changed=PYO3_CROSS_LIB_DIR
        cargo:rerun-if-env-changed=PYO3_CROSS_PYTHON_VERSION
        cargo:rerun-if-env-changed=PYO3_CROSS_PYTHON_IMPLEMENTATION
        cargo:rerun-if-env-changed=PYO3_PRINT_CONFIG
        cargo:rerun-if-env-changed=PYO3_USE_ABI3_FORWARD_COMPATIBILITY

        --- stderr
        error: the configured Python interpreter version (3.13) is newer than PyO3's maximum supported version (3.12)
        = help: please check if an updated version of PyO3 is available. Current version: 0.21.2
        = help: set PYO3_USE_ABI3_FORWARD_COMPATIBILITY=1 to suppress this check and build anyway using the stable ABI
      warning: build failed, waiting for other jobs to finish...
      💥 maturin failed
        Caused by: Failed to build a native library through cargo
        Caused by: Cargo build finished with "exit status: 101": `env -u CARGO MACOSX_DEPLOYMENT_TARGET="11.0" PYO3_ENVIRONMENT_SIGNATURE="cpython-3.13-64bit" PYO3_PYTHON="/Users/admin/projects/kepimetheus/venv/bin/python3" PYTHON_SYS_EXECUTABLE="/Users/admin/projects/kepimetheus/venv/bin/python3" "cargo" "rustc" "--message-format" "json-render-diagnostics" "--manifest-path" "/private/var/folders/_y/z8sthm2j7s9f5rbld7j1xbq00000gp/T/pip-install-_5ecfb_4/promql-parser_59f88903cc0b4cb88976d66a9e022747/Cargo.toml" "--release" "--lib" "--" "-C" "link-arg=-undefined" "-C" "link-arg=dynamic_lookup" "-C" "link-args=-Wl,-install_name,@rpath/promql_parser.cpython-313-darwin.so"`
      Error: command ['maturin', 'pep517', 'build-wheel', '-i', '/Users/admin/projects/kepimetheus/venv/bin/python3', '--compatibility', 'off'] returned non-zero exit status 1
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for promql-parser
Failed to build promql-parser
```

**Comment by cavanierc** [comment]
The version of PyO3 used in the project is too old to build in python 3.13, you should use at least version [0.22.0](https://github.com/PyO3/pyo3/releases/tag/v0.22.0) of PyO3 and it should work. It would certainly have to be changed in Cargo.toml, I don't know if that would have any other impact.

**Comment by messense** [comment]
Python 3.13 support has been released in v0.4.2.
